pkg/cmd/template/schema_test.go

```diff
@@ -14,16 +13,10 @@ import (
 	"github.com/k14s/ytt/pkg/files"
 )
 
-var opts *cmdtpl.Options
-
-func TestMain(m *testing.M) {
-	opts = cmdtpl.NewOptions()
```

Comment:
The way I was thinking of this is: As I am adding tests for `cmd_data_values_test` I want to follow the pattern that we began with `schema_test` like using sub-tests. I am not able to use TestMain in `cmd_data_values_test` to do test setup since it's package public, so I deleted it from schema since we cannot use TestMain as a pattern for all tests. 

It does add duplicate code, but we do need different opts flag setup for tests, take [this schema disabled line 691](https://github.com/vmware-tanzu/carvel-ytt/pull/301/files#diff-2ab902ad23ffe96207ff6466403834df5538e509639024bfc503aed26a377cb8R691) and [this schema enabled line 598](https://github.com/vmware-tanzu/carvel-ytt/pull/301/files#diff-2ab902ad23ffe96207ff6466403834df5538e509639024bfc503aed26a377cb8R597-R598) test. It also shows the prerequisites for each test group in an easy to read way. 

An alternative way we could do this is a setup() function but then we must return the opts. I'm open to thoughts on this.